### PR TITLE
zylProg: made manager static

### PR DESCRIPTION
**Pull request by nandbert**
... so it works without the need of creating an instance first.

P.S. I only tested if it compiles, not if it actually works like this :-)

### CODE REVIEW

File: include/zylProg.h [hunk 1]
```diff
@@ -34,46 +40,61 @@ class zylProg{
...
+	static void					composite(CRGB *fb);
+	static void					input(uint8_t x, uint8_t y, uint8_t z);
 };
 
-void zylProgManager::add(zylProg* ptr){
-	if (m_pHead == NULL){ //first program
-		m_pHead = ptr;
-		m_pActive = ptr;
+
+/* move to cpp *****************/
```
Comment (kwautztretschke) [inline include/zylProg.h:55]: und los :)

**Review by kwautztretschke — CHANGES_REQUESTED** [review]
Schaumermal

File: include/zylProg.h [hunk 2]
```diff
@@ -34,46 +40,61 @@ class zylProg{
...
+int zylProgManager::g_Count=0;
+zylProg *zylProgManager::g_pHead=NULL;
+zylProg *zylProgManager::g_pActive=NULL;
+
+// implement "static methods" (i.e. global functions)
+
+void zylProgManager::add(zylProg* ptr)
+{
+	if (g_pHead == NULL){ //first program
+		g_pHead = ptr;
```
Comment (nandbert) [inline include/zylProg.h:68]: P.S. If you need to execute some code to init zylProgManager (like in a real constructor) you could add it here

File: include/zylProg.h (hunk 1 above)
Comment (nandbert) [reply include/zylProg.h:55]: Keine Email :-(

File: include/zylProg.h [hunk 3]
```diff
@@ -34,46 +40,61 @@ class zylProg{
...
 		ptr = ptr->m_pNext;
-	m_pActive = ptr;
-	m_pActive->activate();
+	g_pActive = ptr;
+	g_pActive->activate();
 }
 
-void zylProgManager::input(uint8_t x, uint8_t y, uint8_t z){
-	m_pActive->input(x, y, z);
+void zylProgManager::input(uint8_t x, uint8_t y, uint8_t z) // I would strongly recommend int x,
```
[the remark this answers is not in the document]
Comment (nandbert) [reply include/zylProg.h:88]: Dir gibbi glei ...

**Review by kwautztretschke — APPROVED** [review]
So, basd

**Pull request merged**